app/javascript/components/user-form/index.jsx

```diff
@@ -0,0 +1,229 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import MiqFormRenderer from '@@ddf';
+import { Loading } from 'carbon-components-react';
+import createSchema from './user-form.schema';
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+import { API } from '../../http_api';
+
+const UserForm = ({ id, data }) => {
+  const [{ initialValues, isLoading }, setState] = useState({ isLoading: true });
+  const [editMode, setEditMode] = useState(false);
+  const [groups, setGroups] = useState(undefined);
```

Comment:
we can integrate these into a single object so that multiple rendering can be avoided on functions like `onFormReset`

```
const [data, setData] = useState({
    initialValues: undefined,
    isLoading: true,
    editMode: false,
    groups: undefined,
  });
```